### PR TITLE
WIP: Add ability to list images on push

### PR DESCRIPTION
**Pull request by saschagrunert**
Due to containers-storage enhancements we're now able to utilize the read-only API for different use cases. One use case is listing container images during a push, like:

```
> podman run -d --rm --name=registry -p 5000:5000 registry
> podman pull quay.io/crio/nix:1.1.0
> podman tag quay.io/crio/nix:1.1.0 localhost:5000/nix
> podman push --tls-verify=false localhost:5000/nix
```

Open a second terminal during the push operation (should take some seconds depending  on the hardware) and try it out:
```
> podman images
REPOSITORY                    TAG         IMAGE ID       CREATED          SIZE
…
```

Podman is not blocking any more during a pull operation because listing container images needs only a read-only store.

Needs https://github.com/containers/storage/pull/473

### CODE REVIEW

**Comment by baude** [comment]
when you are closer to merging this, can you elaborate on the commit message?

**Comment by saschagrunert** [comment]
> when you are closer to merging this, can you elaborate on the commit message?

Yes, I'm just trying to figure out if we can do this automatically without letting libpod manually indicate that we need a "read only layer store only".

**Comment by rhatdan** [comment]
@saschagrunert Still working on this one?

**Comment by saschagrunert** [comment]
> @saschagrunert Still working on this one?

Yes it's currently blocked by https://github.com/containers/storage/pull/493 and https://github.com/containers/storage/pull/473 where I'm currently not sure why it is failing here: https://github.com/containers/libpod/pull/4696 :thinking:

**Comment by saschagrunert** [comment]
Ready for review. PTAL @rhatdan @vrothberg @baude 

I also updated the PR body with an example. :)

File: go.mod [hunk 1]
```diff
@@ -82,3 +82,5 @@ require (
 	k8s.io/client-go v0.0.0-20190620085101-78d2af792bab
 	k8s.io/utils v0.0.0-20190607212802-c55fbcfc754a // indirect
 )
+
+replace github.com/containers/storage => github.com/saschagrunert/storage v1.12.3-0.20200206090634-7fe9bfda9d55
```
Comment (vrothberg) [inline go.mod]: Please make it `WIP` PR again to prevent this from being merged accidentally.

Comment (saschagrunert) [reply go.mod]: Sure, sorry about that :+1:

File: cmd/podman/libpodruntime/runtime.go [hunk 2]
```diff
@@ -15,30 +15,35 @@ import (
 
 // GetRuntimeMigrate gets a libpod runtime that will perform a migration of existing containers
 func GetRuntimeMigrate(ctx context.Context, c *cliconfig.PodmanCommand, newRuntime string) (*libpod.Runtime, error) {
-	return getRuntime(ctx, c, false, true, false, true, newRuntime)
+	return getRuntime(ctx, c, false, true, false, false, true, newRuntime)
```
Comment (rhatdan) [inline cmd/podman/libpodruntime/runtime.go]: I think it is time to make getRuntime take a structure, real tough to read this and understand what is going on, for future users.

Comment (saschagrunert) [reply cmd/podman/libpodruntime/runtime.go]: I do this in https://github.com/containers/libpod/pull/5126 and will rebase later on top here if merged.

**Comment by saschagrunert** [comment]
Green :heavy_check_mark:

**Comment by mheon** [comment]
OK to drop the WIP?

**Comment by saschagrunert** [comment]
> OK to drop the WIP?

Unfortunately not, since we still have to 
```
replace github.com/containers/storage => github.com/saschagrunert/storage v1.12.3-0.20200211092915-485fad532f1b
```
for demonstration purposes.

**Comment by mheon** [comment]
Ack, so we're holding on storage changes - SGTM. I'll take a look.

File: test/system/050-stop.bats [hunk 3]
```diff
@@ -57,8 +57,8 @@ load helpers
 
         # The 'stop' command should return almost instantaneously
         delta_t=$(( $t1 - $t0 ))
-        [ $delta_t -le 2 ] ||\
-            die "podman stop: took too long ($delta_t seconds; expected <= 2)"
+        [ $delta_t -le 3 ] ||\
+            die "podman stop: took too long ($delta_t seconds; expected <= 3)"
```
Comment (vrothberg) [inline test/system/050-stop.bats]: The changes here seem unrelated and should be in a separate commits explaining why they're needed.

**Comment by mheon** [comment]
Same, LGTM once the storage PR goes green

File: test/system/050-stop.bats (hunk 3 above)
Comment (saschagrunert) [reply test/system/050-stop.bats]: Honestly I have no idea, do you saw this in other PRs failing as well? I constantly had a higher delay in that test... :man_shrugging:

**Comment by rhatdan** [comment]
@saschagrunert Any movement on this?

**Comment by saschagrunert** [comment]
> @saschagrunert Any movement on this?

Still depends on https://github.com/containers/storage/pull/473, I'll ping there.
